plugins/catalog-backend/src/next/ConfigLocationEntityProvider.ts

```diff
@@ -15,27 +15,49 @@
  */
 
 import { Config } from '@backstage/config';
+import { ObservableConfig } from '@backstage/backend-common';
 import path from 'path';
 import { getEntityLocationRef } from './processing/util';
 import { EntityProvider, EntityProviderConnection } from './types';
 import { locationSpecToLocationEntity } from './util';
 
 export class ConfigLocationEntityProvider implements EntityProvider {
-  private connection: EntityProviderConnection | undefined;
-
-  constructor(private readonly config: Config) {}
+  constructor(private readonly config: Config | ObservableConfig) {}
```

Comment:
One more comment I guess. It's pretty unfortunate to have these two separate interfaces and do feature detection in code. I feel it would be pretty desirable to try to unify. To have the change subscription in `Config` (and some implementations just won't call it), remove `ObservableConfig`, and have like a `SettableConfig` concrete class.

Reply:
Hmm, yeah that fure sure makes things a bit cleaner. I wanna be pretty confident about the interface if we do that tho, perhaps make it experimental. It's a bit confusing the no frontend code will be able to use the subscribe method though, and it's also a bit tricky to provide feedback when that's done incorrectly. Might be enough with docs though, and tbh I'm thinking it might even be possible to make it available in the frontend in the future.

Reply:
Yeah I think I still vote my suggestion despite the drawbacks. Hm

Reply:
Are you thinking as a breaking change as well? Or making the subscribe method optional or smth like that? Tbh that might make sense anyway

Reply:
Optional may look a bit odd perhaps long term but yeah that's probably best. It does signal the "may or may not be implemented" well, after all. Luckily it's just one method - if it was a more complex interface it might start to get more hairy

Reply:
Yeah could consider something like

```ts
interface Config {
  watch?: {
    subscribe: ...
  }
  ...
}
```

Reply:
So gonna refactor to add `subscribe?: ...` to the `Config` interface, but also try to get rid of the proxy class and just build this all straight into the `ConfigReader` instead, before merging